backend/spec/services/spreadsheet_generator_spec.rb

```diff
@@ -0,0 +1,64 @@
+require "roo"
+require "tempfile"
+
+RSpec.describe SpreadsheetGenerator do
+  describe "#perform" do
+    let!(:enrollment) { create(:enrollment, :franceconnect) }
+    let!(:foreign_enrollment) { create(:enrollment, :api_entreprise) }
+    let(:enrollments) { [enrollment, foreign_enrollment] }
+
+    let(:spreadsheet_generator) { SpreadsheetGenerator.new(enrollments) }
+    let(:file) { Tempfile.new("spreadsheet") }
+
+    context "when enrollments present" do
+      it "generates a spreadsheet with an header row" do
+        result = spreadsheet_generator.perform
+        file.binmode
+        file.write(result)
+        file.rewind
+        workbook = Roo::Spreadsheet.open(file.path, extension: :xlsx)
+
+        headers = %w[id target_api created_at updated_at status organization_id siret nom_raison_sociale zip_code
+          technical_team_type technical_team_value demarche intitule description type_projet
+          date_mise_en_production volumetrie_approximative scopes data_recipients data_retention_period
+          data_retention_comment fondement_juridique_title fondement_juridique_url demandeur_email
+          team_members_json cgu_approved dpo_is_informed additional_content linked_token_manager_id
+          previous_enrollment_id copied_from_enrollment_id link]
+
+        expect(workbook).not_to be_nil
+        expect(workbook.sheet(0).row(1)).to eq(headers)
+
+        file.unlink
+      end
+
+      it "generates a spreadsheet with enrollments" do
+        result = spreadsheet_generator.perform
+        file.binmode
+        file.write(result)
+        file.rewind
+        workbook = Roo::Spreadsheet.open(file.path, extension: :xlsx)
+
+        expect(workbook.sheet(0).row(2) && workbook.sheet(0).row(3)).not_to be nil
+        expect(workbook.sheet(0).row(2)[0]).to eq(enrollments[0].id)
+        expect(workbook.sheet(0).row(3)[0]).to eq(enrollments[1].id)
+
+        file.unlink
+      end
```

Comment:
idem

```suggestion
      it "generates a spreadsheet with enrollments" do
        expect(spreadhseet.sheet(0).row(2)[0]).to eq(enrollments[0].id)
        expect(spreadhseet.sheet(0).row(3)[0]).to eq(enrollments[1].id)
      end
```

tu pourrais aussi compter le nombre de row (je sais pas comment faire) histoire d'être sûre.